### DISCUSSION TITLE
Import of LUT tables

### DISCUSSION THREAD

**Discussion by chbartsch** [awawa-dev/HyperHDR#227] [Q&A]
Hi, yesterday I tried to import a custom LUT table but had no success. Or let's say it was imported but turned the image totally blue.

I created the LUT with Affinity Photo (Photoshop equivalent). Just a tone curve layer and then exported it as a 126^3 3dl-LUT-file.

Does the LUT have to be of any specific size to be imported? Couldn't find an answer in the sources unfortunately.

Thanks!

**Comment by awawa-dev** [comment]
Hi
The import mechanism is starting here https://github.com/awawa-dev/HyperHDR/blob/91f12c1febaca2e7414b4efa1d72e4ff24562a62/assets/webconfig/content/lut_generator.html#L107 probably it needs to be debug in browser's javascript console for reading the header and and least few nodes to see if the format is what it expected. Maybe the problem could be in color scaling, some programs exports nonstandard LUT tables with RGB color scaling above 4096 . There is no limitation for cubes number. Did you import LUT table and then just save it? Imported LUT tables don't need to apply 'generate' button action.

**Reply by chbartsch** [reply]
Thanks for your reply! I was able to solved the problem. There is actually a bug in Affinity's LUT exporter: Despite the LUT size you choose there are always 17 values (= their default LUT size) in the header. After manually changing these to 126 values everything works now.

This is so cool. Now I can create LUTs exactly as I want them. :D It's getting better and better... :)
